### PR TITLE
Improve tests for building and command line applications

### PR DESCRIPTION
**Pull request by GernotMaier**
Add tests to prevent bugs encountered while working on deploy:

- run a python build to ensure that a simtools package can be build (this would have caught an error of listing the readme in pyproject.toml as README.rst, while we actually have a README.md)
- run each application / command line tool defined in pyproject.toml (this would have caught an error that an application name included a typo)

### CODE REVIEW

**Comment by GernotMaier** [comment]
Remove the line `scripts.simtools-production = "simtools.applications.production:main` from pyproject.toml. There is no application called production (it is simulate_prod).

**Comment by GernotMaier** [comment]
Same for `scripts.simtools-simulate-showers-for-trigger-rates`, this application has been removed (simulate_prod provides the same functionality)

File: .github/workflows/CI-integrationtests.yml [hunk 1]
```diff
@@ -19,7 +19,50 @@ on:
...
+      options: --user 0
+
+    steps:
+
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: Test command line tools
+        run: |
+          # read list of scripts from pyproject.toml and execute them
```
Comment (tobiaskleiner) [inline .github/workflows/CI-integrationtests.yml:57]: Should we add `set -e` here?

Comment (GernotMaier) [reply .github/workflows/CI-integrationtests.yml:58]: OK, added.

**Review by tobiaskleiner — APPROVED** [review]

**Pull request merged**